lib/src/default_index/store.rs

```diff
@@ -142,7 +146,11 @@ impl DefaultIndexStore {
         let operations_dir = self.dir.join("operations");
         let commit_id_length = store.commit_id_length();
         let change_id_length = store.change_id_length();
-        let mut new_heads = view.heads().clone();
+        let mut visited_heads: HashSet<CommitId> = view.heads().clone();
+        let mut historical_heads: Vec<(CommitId, OperationId)> = visited_heads
```

Comment:
It's quite a big semantic difference between "new" and "historical" :) But I can see how both make sense from different perspectives. (Not asking for any change)